spec/datadog_spec.rb

```diff
@@ -355,11 +355,9 @@
     end
 
     it 'fails when no application key is provided' do
-      @handler.config[:application_key] = nil
-
       # TODO: figure out how to capture output of Chef::Log
       # Run the report, catch the error
-      expect { @handler.run_report_unsafe(@run_status) }.to raise_error
+      expect {  Chef::Handler::Datadog.new(api_key: API_KEY, application_key: nil) }.to raise_error
```

Comment:
nevermind, got it, it's because the api_key/application_key validation is done on the handler's initialization now